crates/api/src/transport.rs

```diff
@@ -511,3 +546,178 @@ pub struct TransportConnectionStats {
     /// True if this connection has successfully upgraded to webrtc.
     pub is_webrtc: bool,
 }
+
+/// Struct that encapsulates bandwidth usage data for sent and received bytes.
+#[derive(Debug, Clone)]
+pub struct BandwidthStats {
+    bytes_sent: u64,
+    bytes_received: u64,
+}
+
+impl BandwidthStats {
+    /// Creates a new `BandwidthStats` instance with zeroed counters.
+    fn new() -> Self {
+        BandwidthStats {
+            bytes_sent: 0,
+            bytes_received: 0,
+        }
+    }
+
+    /// Increments the sent byte counter.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - The number of bytes sent.
+    pub fn add_sent(&mut self, bytes: u64) {
+        self.bytes_sent += bytes;
+    }
+
+    /// Increments the received byte counter.
+    ///
+    /// # Arguments
+    ///
+    /// * `bytes` - The number of bytes received.
+    pub fn add_received(&mut self, bytes: u64) {
+        self.bytes_received += bytes;
+    }
+
+    /// Returns the total number of bytes sent.
+    pub fn bytes_sent(&self) -> u64 {
+        self.bytes_sent
+    }
+
+    /// Returns the total number of bytes received.
+    pub fn bytes_received(&self) -> u64 {
+        self.bytes_received
+    }
+}
+
+/// `BandwidthTracker` keeps track of bandwidth usage per space and per (space, module) pair.
+pub struct BandwidthTracker {
+    space_stats: Arc<RwLock<HashMap<SpaceId, BandwidthStats>>>,
+    module_stats: Arc<RwLock<HashMap<(SpaceId, String), BandwidthStats>>>,
+}
+
+impl Debug for BandwidthTracker {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BandwidthTracker")
+            .field("space_stats", &self.space_stats)
+            .field("module_stats", &self.module_stats)
+            .finish()
+    }
+}
+
+impl BandwidthTracker {
+    /// Creates a new instance of `BandwidthTracker`.
+    pub fn new() -> Self {
+        Self {
+            space_stats: Arc::new(RwLock::new(HashMap::new())),
+            module_stats: Arc::new(RwLock::new(HashMap::new())),
+        }
+    }
+
+    /// Records data sent for a given space and optional module.
+    ///
+    /// # Arguments
+    ///
+    /// * `space` - The unique identifier for the space.
+    /// * `module` - Optional module identifier. Use `Some(module_id)` for module-specific messages, or `None` for space-level messages.
+    /// * `bytes` - Number of bytes sent.
+    pub fn track_sent(
+        &self,
+        space: SpaceId,
+        module: Option<String>,
+        bytes: u64,
+    ) {
+        if let Ok(mut space_stats) = self.space_stats.write() {
+            space_stats
+                .entry(space.clone())
+                .or_insert_with(BandwidthStats::new)
+                .add_sent(bytes);
+        }
```

Comment:
I suggest to unwrap the write lock:
```suggestion
        self.space_stats.write()
            .unwrap()
            .entry(space.clone())
            .or_insert_with(BandwidthStats::new)
            .add_sent(bytes);
```
That's what we've done throughout this repository with sync Mutexes and RwLocks and what the [examples](https://doc.rust-lang.org/std/sync/struct.RwLock.html#examples) do too.